frontend/src/components/Contributions/ContributionForm.vue

```diff
@@ -88,107 +80,102 @@
-    (!props.isThisMonth && parseInt(form.value.amount) > parseInt(props.maxGddLastMonth))
+// reactive validation schema, because some boundaries depend on form input and existing data
+const validationSchema = computed(() => {
+  const maxAmounts = Number(
+    isThisMonth.value ? parseFloat(props.maxGddThisMonth) : parseFloat(props.maxGddLastMonth),
   )
+  const maxHours = parseFloat(Number(maxAmounts / 20).toFixed(2))
+
+  return object({
+    // The date field is required and needs to be a valid date
+    // contribution date
+    date: dateSchema()
+      .required()
+      .min(new Date(new Date().setMonth(new Date().getMonth() - 1, 1)).toISOString().slice(0, 10)) // min date is first day of last month
+      .max(new Date().toISOString().slice(0, 10))
+      .default(''), // date cannot be in the future
+    memo: memoSchema,
+    hours: number()
+      .required()
+      .transform((value, originalValue) => (originalValue === '' ? undefined : value))
+      .min(0.01, ({ min }) => ({ key: 'form.validation.gddCreationTime.min', values: { min } }))
+      .max(maxHours, ({ max }) => ({ key: 'form.validation.gddCreationTime.max', values: { max } }))
+      .test('decimal-places', 'form.validation.gddCreationTime.decimal-places', (value) => {
+        if (value === undefined || value === null) return true
+        return /^\d+(\.\d{0,2})?$/.test(value.toString())
+      }),
+    amount: number().max(maxAmounts),
+  })
 })
 
-const validMaxGDD = computed(() => {
-  return Number(props.isThisMonth ? props.maxGddThisMonth : props.maxGddLastMonth)
-})
-
-const validMaxTime = computed(() => {
-  return Number(validMaxGDD.value / 20)
-})
+const disabled = computed(() => !validationSchema.value.isValidSync(form))
 
 const noOpenCreation = computed(() => {
   if (props.maxGddThisMonth <= 0 && props.maxGddLastMonth <= 0) {
     return t('contribution.noOpenCreation.allMonth')
   }
-  if (props.isThisMonth && props.maxGddThisMonth <= 0) {
-    return t('contribution.noOpenCreation.thisMonth')
+  if (form.date) {
+    if (isThisMonth.value && props.maxGddThisMonth <= 0) {
+      return t('contribution.noOpenCreation.thisMonth')
+    }
+    if (!isThisMonth.value && props.maxGddLastMonth <= 0) {
+      return t('contribution.noOpenCreation.lastMonth')
+    }
   }
-  if (!props.isThisMonth && props.maxGddLastMonth <= 0) {
-    return t('contribution.noOpenCreation.lastMonth')
-  }
-  return ''
-})
-
-const getMaximalDate = computed(() => {
-  return new Date().toISOString().slice(0, 10)
+  return undefined
 })
 
-watch(
-  () => formValues.hours,
-  () => {
-    updateAmount(formValues.hours)
-  },
-)
-
-function updateAmount(hours) {
-  setFieldValue('amount', (hours * 20).toFixed(2).toString())
+const updateField = (newValue, name) => {
+  if (typeof name === 'string' && name.length) {
+    form[name] = newValue
+    if (name === 'hours') {
+      form.amount = form.hours ? (form.hours * 20).toFixed(2).toString() : '20'
```

Comment:
are the number-values '20' the amount of gradidos per hour?
Isn't it better to define such a global value as a global constant?